src/data/InstanceDhisRepository.ts

```diff
@@ -350,20 +351,24 @@ export class InstanceDhisRepository implements InstanceRepository {
     }
 
     private buildEventsPayload(dataPackage: DataPackage): Event[] {
-        return dataPackage.dataEntries.map(({ id, orgUnit, period, attribute, dataValues, dataForm, coordinate }) => ({
-            event: id,
-            program: dataForm,
-            status: "COMPLETED",
-            orgUnit,
-            occurredAt: period,
-            attributeOptionCombo: attribute,
-            dataValues: dataValues,
-            coordinate: coordinate,
-            geometry: coordinate && {
-                type: "Point",
-                coordinates: [Number(coordinate.longitude), Number(coordinate.latitude)],
-            },
-        }));
+        return dataPackage.dataEntries.map(
+            ({ id, orgUnit, period, attribute, dataValues, dataForm, coordinate, geometry }) => ({
```

Comment:
[best practices] Now that we're refactoring this code, let's take the opportunity to add the return type of the arrow function. This improves code navigation and refactoring: `({ id, orgUnit, period, attribute, dataValues, dataForm, coordinate, geometry }): Event`